tests/testthat/test-printing.R

```diff
@@ -785,3 +786,145 @@ test_that("horizontal separator is propagated from table to print and export", {
   export_txt_tbl <- strsplit(export_as_txt(tbl), "\n")[[1]]
   expect_identical(tostring_tbl, export_txt_tbl)
 })
+
+## higher-level showing ncols works:
+
+test_that("showing higher-level ncols works", {
+  mydat <- subset(ex_adsl, SEX %in% c("M", "F"))
+  mydat$SEX2 <- factor(
+    ifelse(
+      mydat$SEX == "M",
+      "males",
+      "super long sentence that involves females"
+    )
+  )
+
+  lyt <- basic_table() %>%
+    split_cols_by("ARM", show_colcounts = TRUE) %>%
+    split_cols_by("SEX2", show_colcounts = TRUE) %>%
+    split_cols_by("STRATA1") %>%
+    analyze("AGE")
+
+  tbl <- build_table(lyt, mydat)
+  expect_equal(colcount_na_str(tbl), "")
+  colcount_na_str(tbl) <- "wut"
+  expect_equal(colcount_na_str(tbl), "wut")
+  colcount_na_str(tbl) <- ""
+  cwds <- rep(8, ncol(tbl) + 1)
+  expect_equal(nlines(col_info(tbl), colwidths = cwds, fontspec = NULL), 7)
+  mpf <- matrix_form(tbl, TRUE)
+  ## this is to get around complaints about ::: in the precommit rules
+  dcfnw <- get("do_cell_fnotes_wrap", asNamespace("formatters"))
+  mpf <- dcfnw(mpf, cwds, NULL, FALSE, fontspec = NULL)
+  strs <- mf_strings(mpf)
+  ## wrapping some cells and not others still works
+  expect_equal(strs[3:4, 2], c("", "males"))
+
+  expect_equal(strs[2, 2], "(N=130)")
+  ## N= cells all across rows 2 (for ARM) and 5 (for SEX2), except rowlabels
+  expect_true(all(grepl("(N=", strs[c(2, 5), -1], fixed = TRUE)))
+  ## No N= cells elsewhere
+  expect_true(all(!grepl("(N=", strs[-c(2, 5), -1], fixed = TRUE)))
+
+  broken_tbl <- tbl
+  expect_true(colcount_visible(broken_tbl, c("ARM", "A: Drug X", "SEX2", "males")))
+  colcount_visible(broken_tbl, c("ARM", "A: Drug X", "SEX2", "males")) <- FALSE
+  expect_error(print(broken_tbl), "different colcount visibility among sibling facets")
+
+  ## does the old accessor still work ok
+
+  lyt2 <- basic_table() %>%
+    split_cols_by("ARM", show_colcounts = TRUE) %>%
+    split_cols_by("SEX2", show_colcounts = TRUE) %>%
+    split_cols_by("STRATA1", show_colcounts = TRUE) %>%
```

Comment:
Nice!! multiple values too!! Also with different formats?

Reply:
yes, but currently not different na strings.